zubhub_backend/zubhub/projects/tests.py

```diff
@@ -1,3 +1,139 @@
+        request.user = user
+
+        response = ProjectCreateAPIView.as_view()(request)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+    def test_create_project_unauthenticated_user(self):
+        data = {
+            "title": "Test Project",
+            "description": "This is a test project",
+            "images": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRFNzTqQLNPCOdIgNJIIYyb6ovFd_w0ZJ9eJA&usqp=CAU",
+            "video": "https://youtu.be/ol_DAvdpigY?si=dZupzpRGIige8Crp",
+            "materials_used": "Tape, Wire",
+            "category": "Robotics",
+            "publish": {"type": 4, "visible_to": []}
+        }
+        request = self.factory.post('/projects/create/', data)
+        request.user = AnonymousUser()
+
+        response = ProjectCreateAPIView.as_view()(request)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        # Add more assertions for the response data
+
+    # Add more test cases for other scenarios
+
+
+class ProjectUpdateAPITest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.project = Project.objects.create(title='Test Project')
+
+    def test_update_project_authenticated_user(self):
+        user = User.objects.create(username='testuser')
+        data = {
+            "title": "Test Project",
+            "description": "This is a test project",
+            "images": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRFNzTqQLNPCOdIgNJIIYyb6ovFd_w0ZJ9eJA&usqp=CAU",
+            "video": "https://youtu.be/ol_DAvdpigY?si=dZupzpRGIige8Crp",
+            "materials_used": "Tape, Wire",
+            "category": "Robotics",
+            "publish": {"type": 4, "visible_to": []}
+        }
+        request = self.factory.patch(f'/projects/{self.project.pk}/', data)
+        request.user = user
+
+        response = ProjectUpdateAPIView.as_view()(request, pk=self.project.pk)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_project_unauthenticated_user(self):
+        data = {
+            "title": "Test Project",
+            "description": "This is a test project",
+            "images": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRFNzTqQLNPCOdIgNJIIYyb6ovFd_w0ZJ9eJA&usqp=CAU",
+            "video": "https://youtu.be/ol_DAvdpigY?si=dZupzpRGIige8Crp",
+            "materials_used": "Tape, Wire",
+            "category": "Robotics",
+            "publish": {"type": 4, "visible_to": []}
+        }
+        request = self.factory.patch(f'/projects/{self.project.pk}/', data)
+        request.user = AnonymousUser()
+
+        response = ProjectUpdateAPIView.as_view()(request, pk=self.project.pk)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+class ProjectDeleteAPITest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.project = Project.objects.create(title='Test Project')
```

Comment:
I think we should make the project belong to a particular user. That way we can test if a user can delete another user's project.